sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -197,12 +227,19 @@ object RmmRapidsRetryIterator extends Arm with Logging {
     try {
       fn
     } catch {
-      case t: RetryOOM =>
-        r.restore()
-        throw t
-      case t: SplitAndRetryOOM =>
-        r.restore()
-        throw t
+      case ex: Throwable =>
+        // Only restore on retry exceptions
+        val (topLevelIsRetry, _) = isRetryOrSplitAndRetry(ex)
+        val causedByRetry = topLevelIsRetry match {
```

Comment:
given the or in line 239, I don't think this match is necessary.
```
val (causedByRetry, _) = causedByRetryOrSplit(ex)
```

Then the the or is "top level OR caused", so this would be equivalent.

Reply:
I was trying to avoid calling `causedByRetryOrSplit` unless needed.  Does the OR accomplish this by itself?  It's a light enough function that it is probably worth just calling it to make code simplet.

Reply:
good point, fine not to. It's fine not to change this it was just a nit.